### PR TITLE
Mipmap coverage clean

### PR DESCRIPTION
**Pull request by fcollman**
in an effort to improve our code coverage i discovered some inconsistencies in the modules, mostly around exception handling and validation that wasn't being tested.  This PR cleans up some of those problems and improves the testing coverage.

### CODE REVIEW

File: rendermodules/dataimport/generate_mipmaps.py [hunk 1]
```diff
@@ -98,17 +98,16 @@ def run(self):
             zvalues1 = range(self.args['zstart'], self.args['zend'] + 1)
             # extract only those z's that exist in the input stack
             zvalues = list(set(zvalues1).intersection(set(zvalues)))
-        except NameError:
+        except KeyError:
             try:
                 if self.args['z'] in zvalues:
                     zvalues = [self.args['z']]
```
Comment (RussTorres) [inline rendermodules/dataimport/generate_mipmaps.py]: If you pick a z outside of this zvalues in the stack, does this generate mipmaps for the entire stack?

Comment (mahaling) [reply rendermodules/dataimport/generate_mipmaps.py]: The zvalues list should be empty in that case and hence the code should exit gracefully without generating any mipmaps

Comment (fcollman) [reply rendermodules/dataimport/generate_mipmaps.py]: Russel is right.. if you only put self.args['z'] and that z value isn't in the stack, then the if statement isn't called, and zvalues remains all the zvalues which isn't the right behavior.  I'll fix it with an else statement and add a test.

Comment (mahaling) [reply rendermodules/dataimport/generate_mipmaps.py]: ok got it.... I was referring to line 100...

**Pull request merged**